### PR TITLE
CI: INFRA-2087 auto generated issues for mobile and extension repositories

### PR DESCRIPTION
**Pull request by rrabenda**
## Explanation

CI pipeline for auto generated issues when major release of any package occurs. New issues will be created in metamask-extension and metamask-mobile repositories.

## References

* Related to https://github.com/MetaMask/MetaMask-planning/issues/2988

## Changelog

None

### CODE REVIEW

File: .github/workflows/create-update-issues.yaml
Comment (rrabenda) [inline .github/workflows/create-update-issues.yaml:24]: I'm not 100% sure that we have to fetch here whole history, if release pipeline have always tag on head commit `fetch-tags: true` to make it little faster

File: .github/workflows/main.yml
Comment (rrabenda) [inline .github/workflows/main.yml:58]: Before merge security team have to create new github app, I will send request to @NicholasEllul

File: .github/workflows/create-update-issues.yaml
Comment (rrabenda) [inline .github/workflows/create-update-issues.yaml:76]: For now I hardcoded repositories names if you see any case that this have to be dynamic I can move this step to new job and leverage workflow matrix.

File: .github/workflows/create-update-issues.yaml
Comment (rrabenda) [inline .github/workflows/create-update-issues.yaml:79]: For now I hardcoded repositories names if you see any case that this have to be dynamic I can move this step to new job and leverage workflow matrix.

Comment (Gudahtt) [reply .github/workflows/create-update-issues.yaml:79]: Hard-coding these seems fine to me 👍

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [inline .github/workflows/create-update-issues.yaml:78]: We'll want to assign teams as well, this was part of the acceptance criteria and is important for achieving the main goal here (visibility and accountability for these updates).

We use labels to keep track of who owns which issue, so assigning a team here means attaching that team's label to the issue (which luckily is easy with the `gh issue` command).

However, determining which team to assign might be tricky. The ticket suggests that we assign it to "the responsible team", but doesn't describe who should be responsible. In discussing this work, we had considered either:
* Assigning the team who released the work on core, or
* Assigning the "codeowner" team(s) for the given package.

I think either way would be fine to start with, as the assigned team can easily re-assign a different team as appropriate. As long as some team is assigned, it accomplishes the goals of visibility and accountability.

For the first strategy, we could get the team label using an approach like this: https://github.com/MetaMask/github-tools/blob/058012b49ff2fbd9649c566ba43b29497f93b21d/.github/workflows/add-team-label.yml#L19
 (we already have a mapping of user to team in our planning repo)
 
 For the second strategy, we could parse the `.github/CODEOWNERS` file. But we'd need a mapping of GitHub team names to labels (the format is a little different between the two right now, and doesn't follow a consistent naming convention from team to team).

**Review by Gudahtt — CHANGES_REQUESTED** [review]

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [reply .github/workflows/create-update-issues.yaml:24]: I think it would be safe just to get the HEAD commit. The release tags will always be on that commit. The tags are created in the `publish-release` step, which happens before this one.

Comment (rrabenda) [reply .github/workflows/create-update-issues.yaml:24]: Could publish release output created tags as it have it? This would simplify whole process. We can use also use `git tag --points-at head` command and process output it the loop

File: .github/workflows/create-update-issues.yaml
Comment (rrabenda) [reply .github/workflows/create-update-issues.yaml:29]: Could `publish-release` step output created tags as it have it? This would simplify whole process. We can use also use `git tag --points-at head` command and process output it the loop

File: .github/workflows/create-update-issues.yaml
Comment (rrabenda) [reply .github/workflows/create-update-issues.yaml:61]: We use GitHup app here to avoid personal access token. PAT is connected to user and when that user will lost access to org also workload will stop working. On top off that is easier to rotate credentials for security team if required.

Comment (Gudahtt) [reply .github/workflows/create-update-issues.yaml:61]: We have a fairly well defined process for managing PATs, and we generate them using an account not tied to an individual (`metamaskbot`). We use them a lot, it would be quite onerous to create an app for each one.

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [reply .github/workflows/create-update-issues.yaml:29]: I don't think that would be easily accomplished in this case, because of how the publish workflow/action is setup.

However, getting all tags on the HEAD commit is quite easy: `git describe --tags --match '@metamask/*' HEAD`

File: teams.json
Comment (rrabenda) [inline teams.json:26]: I wans't able to find `@MetaMask/product-safety` team label. @Gudahtt Could you provide it and double check if rest of mapping is correct?

Comment (Gudahtt) [reply teams.json:26]: It's `team-product-safety`. They were missing from the JSON file I sent you (they were just added to that today)

File: teams.json
Comment (Gudahtt) [inline teams.json:1]: Thoughts on mapping each controller to an array of team labels, rather than just one? Many of them have multiple code owners. Choosing just one for the label seems a bit arbitrary.

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [inline .github/workflows/create-update-issues.yaml:7]: This adds more specifics on the required permissions, and fixes a typo:
```suggestion
        description: Github token with permission to create issues in both mobile and extension repositories
```

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [inline .github/workflows/create-update-issues.yaml:45]: Nit: Add trailing newline
```suggestion
        shell: bash

```

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [inline .github/workflows/create-update-issues.yaml:13]: Nit: Read permissions are the default, I would not expect this to be required

```suggestion
```

**Review by makemesteaks — DISMISSED** [review]

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [inline .github/workflows/create-update-issues.yaml:36]: Typo:
```suggestion
                # Fetch responsible team from file
```

File: .github/workflows/create-update-issues.yaml
Comment (Gudahtt) [inline .github/workflows/create-update-issues.yaml:39]: Update repositories to real targets
```suggestion
                gh issue create --title "Update ${package_name} to version ${version}" --body "Please update ${package_name} to version ${version}" --repo "MetaMask/metamask-extension" --label "$teams"
                gh issue create --title "Update ${package_name} to version ${version}" --body "Please update ${package_name} to version ${version}" --repo "MetaMask/metamask-mobile" --label "$teams"
```

File: teams.json
Comment (Gudahtt) [reply teams.json:1]: Done in https://github.com/MetaMask/core/pull/4763/commits/ad9b71191167c27703146b24c16f8248b5177c18

**Review by Gudahtt — COMMENTED** [review]
LGTM! Though I'll find another reviewer, since I've committed directly to this branch.

**Review by cryptodev-2s — APPROVED** [review]
Looks great! Excited to see this feature go live.

**Review by mcmire — APPROVED** [review]
Looks good!

**Pull request merged**
